app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PasswordResetsController < ApplicationController
+  before_action :load_user, :valid_user, :check_expiration,
+    only: [:edit, :update]
+
+  def new; end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t ".find"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t ".not_find"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].blank?
+      @user.errors.add(:password, t(".password_error"))
+      render :edit
+    elsif @user.update_attributes(user_params)
+      log_in @user
+      flash[:success] = t ".done"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:password, :password_confirmation)
+  end
+
+  def load_user
+    @user = User.find_by(id: params[:id])
+    return if @user
+    flash[:danger] = t ".user_notfound"
+    redirect_to root_path
+  end
+
+  def valid_user
+    unless @user&.activated? &&
```

Comment:
có cần `@user&.activated?` ko chú, hay chỉ cần `@user.activated?`

khi nào mình nên sử dụng phép toán an toàn `&.` nhi?

Reply:
Dạ chị là không cần ạ. Vì khi làm e chỉ nghĩ là làm đúng logic nên ghi ra cho tường minh :3 .
sử dụng &. khi mình gọi một phương thức hay 1 helper nào đó mà user nill thì nó báo ngoại lệ nên mình phải check user tồn tại trước